engine/api/application/dao_application_integration.go

```diff
@@ -0,0 +1,168 @@
+
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+type dbIntegration sdk.IntegrationConfig
+
+// application_deployment_strategy
+type dbApplicationDeploymentStrategy struct {
+	gorpmapping.SignedEntity
+	ID                   int64         `db:"id"`
+	ProjectIntegrationID int64         `db:"project_integration_id"`
+	ApplicationID        int64         `db:"application_id"`
+	Config               dbIntegration `db:"cipher_config" gorpmapping:"encrypted,ProjectIntegrationID,ApplicationID"` //config
+}
+
+func (e dbApplicationDeploymentStrategy) Canonical() gorpmapping.CanonicalForms {
+	var _ = []interface{}{e.ProjectIntegrationID, e.ApplicationID}
+	return gorpmapping.CanonicalForms{
+		"{{print .ProjectIntegrationID}}{{print .ApplicationID}}",
+	}
+}
+
+func newDBApplicationDeploymentStrategy(projectIntegrationID, applicationID int64) *dbApplicationDeploymentStrategy {
+	return &dbApplicationDeploymentStrategy{
+		ProjectIntegrationID: projectIntegrationID,
+		ApplicationID:        applicationID,
+	}
+}
+
+func (e *dbApplicationDeploymentStrategy) SetConfig(cfg sdk.IntegrationConfig) {
+	e.Config = dbIntegration(cfg)
+}
+
+func (e *dbApplicationDeploymentStrategy) IntegrationConfig() sdk.IntegrationConfig {
+	return sdk.IntegrationConfig(e.Config).Clone()
+}
+
+// LoadDeploymentStrategies loads the deployment strategies for an application
+func LoadDeploymentStrategies(db gorp.SqlExecutor, appID int64, withClearPassword bool) (map[string]sdk.IntegrationConfig, error) {
+	query := gorpmapping.NewQuery(`SELECT application_deployment_strategy.*
+	FROM application_deployment_strategy
+	WHERE application_deployment_strategy.application_id = $1`).Args(appID)
+
+	var res []dbApplicationDeploymentStrategy
+	if err := gorpmapping.GetAll(context.Background(), db, query, &res, gorpmapping.GetOptions.WithDecryption); err != nil {
+		return nil, sdk.WrapError(err, "unable to load deployment strategies")
+	}
+
+	deps := make(map[string]sdk.IntegrationConfig, len(res))
+	for _, r := range res {
+		isValid, err := gorpmapping.CheckSignature(r, r.Signature)
+		if err != nil {
+			return nil, err
+		}
+		if !isValid {
+			log.Error(context.Background(), "application.LoadDeploymentStrategies> application_deployment_strategy %d data corrupted", appID)
+			continue
+		}
+
+		//Parse the config and replace password values by place holder if !withClearPassword
+		newCfg := sdk.IntegrationConfig{}
+		for k, v := range r.IntegrationConfig() {
+			if v.Type == sdk.IntegrationConfigTypePassword {
+				if !withClearPassword {
+					newCfg[k] = sdk.IntegrationConfigValue{
+						Type:  sdk.IntegrationConfigTypePassword,
+						Value: sdk.PasswordPlaceholder,
+					}
+					continue
+				}
+				newCfg[k] = v
+			} else {
+				newCfg[k] = v
+			}
+		}
+		// Sorry about that :(
+		projectIntegrationName, err := db.SelectStr("SELECT name FROM project_integration WHERE id = $1 ", r.ProjectIntegrationID)
```

Comment:
Maybe we can do without this, anyway can you JOIN directly in the main query to exec only on query?

Reply:
nope because the entity dbApplicationDeploymentStrategy doesn't contains the projectIntegrationName.
projectIntegrationName is a key of the returned map